addons/website/static/src/snippets/s_popup/000.js

```diff
@@ -8,30 +8,46 @@ const utils = require('web.utils');
 
 const PopupWidget = publicWidget.Widget.extend({
     selector: '.s_popup',
-    events: {
+    read_events: {
         'click .js_close_popup': '_onCloseClick',
         'hide.bs.modal': '_onHideModal',
         'show.bs.modal': '_onShowModal',
     },
+    disabledInEditableMode: false,
 
     /**
      * @override
      */
     start: function () {
-        this._popupAlreadyShown = !!utils.get_cookie(this.$el.attr('id'));
-        if (!this._popupAlreadyShown) {
-            this._bindPopup();
+        if (this.editableMode) {
+            this.$target[0].dataset.popupParent = this.$target[0].closest('footer') ?
+                'footer' : 'main';
```

Comment:
99% sure this is wrong, but without any comment, I can't guess. Why would a public widget write on the DOM in edit mode only, especially if this is to remove the attribute in destroy anyway? :thinking:  Can't you use a variable instead of writing the DOM? :thinking: EDIT: understood, that's because your snippet options relies on it... why won't you add a comment to explain that? :/ There must be a better way organisation for all of this, what about:
- Edit mode or not, same code: you check `initialParentEl = ...` in the start and move it back in the destroy
- The option code is in charge of adding the popup at the right initial parent in the first place
- As a stable fix, you can mark the DOM dataset with the value of initialParentEl and use it in the option JS

Something like that, with comments explaining it.